### PR TITLE
Add axes (optional) input to Pad

### PR DESCRIPTION
**Pull request by attila-dusnoki-htec**
Pad version 18 introduced axes, as a way to make pads easier to use. With that, you only need to define the actually changing pads and the axes that they will be used on.

This change only affects the way the parsing works, no modification on the migraphx pad op needed.

Note: no onnx python unittest to enable, because they fail with `PARSE_PAD: pad input must be constant`

Resolves https://github.com/migraphx-benchmark/AMDMIGraphX/issues/71

### CODE REVIEW

File: src/onnx/parse_pad.cpp [hunk 1]
```diff
@@ -182,6 +182,53 @@ struct parse_pad : op_parser<parse_pad>
...
+
+            std::vector<int64_t> new_pads(input_rank * 2);
+            for(size_t idx{0}; idx < num_axes; ++idx)
+            {
+                int64_t axis = axes[idx];
+                // axis can be negative
+                if(axis < 0)
+                {
+                    axis = input_rank + axis;
+                }
```
Comment (umangyadav) [inline src/onnx/parse_pad.cpp:217]:
```suggestion
          int64_t axis = axes[idx] < 0 ? input_rank + axes[idx] : axes[idx];
```

File: test/onnx/onnx_test.cpp [hunk 2]
```diff
@@ -4573,6 +4584,48 @@ TEST_CASE(pad_3arg_test)
...
+    mm->add_literal({migraphx::shape{migraphx::shape::int32_type, {2}}, {1, 3}});
+    // constant_value=1
+    mm->add_literal({migraphx::shape{migraphx::shape::float_type}, {1.0f}});
+    // pads=[1,3,2,4]
+    mm->add_literal({migraphx::shape{migraphx::shape::int32_type, {4}}, {1, 3, 2, 4}});
+    auto r = mm->add_instruction(
+        migraphx::make_op("pad", {{"pads", {0, 1, 0, 3, 0, 2, 0, 4}}, {"value", 1.0f}}), l0);
+    mm->add_return({r});
+
+    std::cout << "mm\n" << *mm << std::endl;
```
Comment (umangyadav) [inline test/onnx/onnx_test.cpp:4602]: Remove prints

**Review by umangyadav — APPROVED** [review]

File: src/onnx/parse_pad.cpp [hunk 3]
```diff
@@ -181,6 +187,81 @@ struct parse_pad : op_parser<parse_pad>
...
+    std::vector<int64_t> parse_axes(const std::vector<instruction_ref>& args,
+                                    bool is_constant_mode) const
+    {
+        std::vector<int64_t> axes{};
+        // axes is 3rd or 4th, depending on constant mode
+        auto pos = is_constant_mode ? 4 : 3;
+        if(args.size() >= pos)
+        {
+            auto axes_arg = args.at(pos - 1)->eval();
+            check_arg_empty(axes_arg, "PARSE_PAD: pad input must be constant");
```
Comment (CharlieL7) [inline src/onnx/parse_pad.cpp:202]:
```suggestion
            check_arg_empty(axes_arg, "PARSE_PAD: variable `axes` input not supported");
```

File: src/onnx/parse_pad.cpp [hunk 4]
```diff
@@ -152,18 +127,49 @@ struct parse_pad : op_parser<parse_pad>
...
+    }
 
+    std::vector<int64_t> parse_pads(const onnx_parser::node_info& info,
+                                    const std::vector<instruction_ref>& args) const
+    {
+        std::vector<int64_t> pads{};
+        if(args.size() >= 2)
+        {
+            auto pad_arg = args.at(1)->eval();
+            check_arg_empty(pad_arg, "PARSE_PAD: pad input must be constant");
```
Comment (CharlieL7) [inline src/onnx/parse_pad.cpp:152]:
```suggestion
            check_arg_empty(pad_arg, "PARSE_PAD: `pads` input must be constant");
```

**Review by CharlieL7 — APPROVED** [review]
Minor suggestions to make throws more informative

**Pull request merged**